### PR TITLE
Clear HttpContext.User when the user is disposed.

### PR DESCRIPTION
**Pull request by Tratcher**
https://github.com/aspnet/Logging/issues/543#issuecomment-321907828

Hosting logs between OnCompleted/RegisterForDispose and DisposeContext. Loggers using IHttpContextAccessor can access User and get ODEs. We can't dispose the user any later, but we can clear it once it's been disposed.

### CODE REVIEW

**Comment by cesarbs** [comment]
Can you add a test?

**Comment by Tratcher** [comment]
Not readily, there are too many layers of the stack involved.

**Comment by cesarbs** [comment]
LGTM, assuming you tested this manually. It would be nice to capture the test scenario somewhere so anyone knows how to verify this.

File: src/Microsoft.AspNetCore.Server.IISIntegration/AuthenticationHandler.cs [hunk 1]
```diff
@@ -51,13 +52,26 @@ private WindowsPrincipal GetUser()
...
 
             return _user;
         }
 
+        private static Task ClearUser(object arg)
+        {
+            var context = (HttpContext)arg;
+            if (context.User is WindowsPrincipal)
+            {
+                context.User = null;
```
Comment (HaoK) [inline src/Microsoft.AspNetCore.Server.IISIntegration/AuthenticationHandler.cs]: Don't we guarantee HttpContext.User to be non null?  Should we just set it to a new ClaimsPrincipal?  Or will the context do that automatically if someone asks for it?

Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IISIntegration/AuthenticationHandler.cs]: DefaultHttpContext handles that. https://github.com/aspnet/HttpAbstractions/blob/5128efe0e88eaa176d4231ff62d754de59bb0edd/src/Microsoft.AspNetCore.Http/DefaultHttpContext.cs#L131-L143

**Pull request merged**